client_test.go

```diff
@@ -322,3 +325,71 @@ func TestBeforeSendGetAccessToEventHint(t *testing.T) {
 
 	assertEqual(t, transport.lastEvent.Message, "customComplexError: Foo 42")
 }
+
+func TestSampleRate(t *testing.T) {
+	tests := []struct {
+		SampleRate float64
+		// tolerated range is [SampleRate-MaxDelta, SampleRate+MaxDelta]
+		MaxDelta float64
+	}{
+		// {0.00, 0.0}, // oddly, sample rate = 0.0 means 1.0, skip test for now.
```

Comment:
It's because of zero-value for floats. I had no idea how to reasonably go around it ¯\_(ツ)_/¯

Reply:
There are a few possibilities like [Functional Options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis); use pointers to detect zero-value vs not-set; etc. All with pros and cons.

Since this is part of `sentry.Init`, any change would break every example code out there :/